forge-game/src/main/java/forge/game/ability/AbilityUtils.java

```diff
@@ -2634,6 +2634,30 @@ public static int xCount(Card c, final String s, final CardTraitBase ctb) {
             return game.getPhaseHandler().getPlanarDiceSpecialActionThisTurn();
         }
 
+        if (sq[0].equals("AllTypes")) {
+            List<Card> cards = getDefinedCards(c, sq[1], ctb);
+
+            int amount = countCardTypesFromList(cards, false) +
+                    countSuperTypesFromList(cards) +
+                    countSubTypesFromList(cards);
+
+            return doXMath(amount, expr, c, ctb);
+        }
+        if (sq[0].contains("CardTypes")) {
```

Comment:
still lots of bad changes in this file that already got refactored
please no changes except related for the new AllTypes property

Reply:
Which ones exactly are you referring to? I'm trying to figure out what is what here tbh and I got quite tangled in these changes :/

Reply:
Ok, removed most of the stuff unrelated to AllTypes, but there's also a rename for get.... -> count.... - should that go as well or should it be kept as countXXX?

Reply:
if it compiles now it's probably fine

Reply:
Yes, tested the compile and the card itself, looks like it works fine. Will merge this one then. Thanks for looking at it! :+1: